tests/t0112_gateway_cors_test.go

```diff
@@ -0,0 +1,52 @@
+package tests
+
+import (
+	"testing"
+
+	"github.com/ipfs/gateway-conformance/tooling/test"
+	. "github.com/ipfs/gateway-conformance/tooling/test"
+)
+
+func TestCors(t *testing.T) {
+	cidHello := "bafkqabtimvwgy3yk" // hello
+
+	tests := SugarTests{
+		{
+			Name: "GET Response from Gateway should contain CORS headers",
+			Request: Request().
+				Path("ipfs/{{CID}}/", cidHello),
+			Response: Expect().
+				Headers(
+					Header("Access-Control-Allow-Origin").Equals("*"),
+					Header("Access-Control-Allow-Methods").Equals("GET"),
+					Header("Access-Control-Allow-Headers").Has("Range"),
+					Header("Access-Control-Expose-Headers").Has(
+						"Content-Range",
+						"Content-Length",
+						"X-Ipfs-Path",
+						"X-Ipfs-Roots",
+					),
+				),
+		},
+		{
+			Name: "OPTIONS to Gateway succeeds",
+			Request: Request().
+				Method("OPTIONS").
+				Path("ipfs/{{CID}}/", cidHello),
+			Response: Expect().
+				Headers(
+					Header("Access-Control-Allow-Origin").Equals("*"),
+					Header("Access-Control-Allow-Methods").Equals("GET"),
+					Header("Access-Control-Allow-Headers").Has("Range"),
+					Header("Access-Control-Expose-Headers").Has(
+						"Content-Range",
+						"Content-Length",
+						"X-Ipfs-Path",
+						"X-Ipfs-Roots",
+					),
+				),
+		},
+	}
+
+	test.Run(t, tests)
+}
```

Comment:
@lidel I believe this is all we want to port from t0112, since the rest is related to the API, but please let me know if we missed something

https://github.com/ipfs/kubo/blob/da28fbc65a2e0f1ce59f9923823326ae2bc4f713/test/sharness/t0112-gateway-cors.sh